### ISSUE TITLE
View annotations disappear with some camera states

### ISSUE THREAD

**Issue by ittna** [mapbox/mapbox-maps-ios#2149]
## Environment

- Xcode version: 15.3
- iOS version: 17.4
- Devices affected: iPhone 15 Pro
- Maps SDK Version: 11.2.0

## Observed behavior and steps to reproduce

In our app we show map pins as view annotations because we want to animate our pins. In tilted map view (pitch > 0) there seems to be bug that with some map rotations view annotations get hidden.

https://github.com/mapbox/mapbox-maps-ios/assets/2304597/5bfd825e-1e81-4fa5-84cf-63cb597a47c2

## Expected behavior

View annotations should be visible when they are in the camera frustum.

## Notes / preliminary analysis

As I understand view annotations are not drawn when they are offscreen. Maybe problem is that camera transformation doesn't correctly project visible area.

**Comment by persidskiy** [comment]
@ittna Do you use camera padding? You can try change the `allowOverlap`/`allowOverlapWithPuck`/`ignoreCameraPadding` https://docs.mapbox.com/ios/maps/api/11.2.0/documentation/mapboxmaps/viewannotation that may help to debug the issue.

**Comment by ittna** [comment]
@persidskiy Yes, I am using camera padding. `ignoreCameraPadding` doesn't fix this issue though. But I think I will use it as it will allow showing view annotations on the edges of the viewport.

Surprisingly `allowOverlapWithPuck` fixes this issue. It's odd as my location puck is in Finland i.e. not anywhere near New York where the my view annotations are. But I guess I should have this on anyway as user could be near my annotations and I want my annotations show in that case also.
